packages/go_router/lib/src/builder.dart

```diff
@@ -70,7 +73,6 @@ class RouteBuilder {
     PopPageCallback onPopPage,
     bool routerNeglect,
   ) {
-    _routeMatchLookUp.clear();
```

Comment:
Yes, that is the contract of `Expando`. As with all things related to CG, it can be hard to verify programmatically though. Perhaps the https://pub.dev/packages/leak_tracker package could be used to verify this (wasn't able to make it work with testWidgets when I tried though).

Reply:
In that case, I think I will prefer using Map for since this map need to be sure to only contain the RouteMatch that are currently in the GoRouter. In case some thing may hold on to the reference for some other reason, especially once we opens up the API

Reply:
The reason for the Expando is for keeping track of Page-to-RouteMatch mappings for cached Navigators in StatefulNavigationShell. Strictly speaking, this caching is more or less an "optimisation", to reduce unnecessary - and sometimes disruptive - rebuilds. However, the other branch (soon to be merged here) contains a new solution for this, that does away with the Expando and moves the Page-to-RouteMatch into a separate class Map (PagePopContext). 

So have a look at that solution and see what you think before you decide. But as I said, StatefulShellRoute would function without this, so we could always remove this and add it in a separate PR. However, leaving this out would mean that there may be layout flashes/glitches in some cases.